flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java

```diff
@@ -197,9 +199,11 @@ protected SecurityContext installSecurityContext(Configuration configuration) th
 		return SecurityUtils.getInstalledContext();
 	}
 
-	private void runCluster(Configuration configuration) throws Exception {
+	private void runCluster(Configuration configuration, PluginManager pluginManager) throws Exception {
 		synchronized (lock) {
-			initializeServices(configuration);
+
+			//TODO: Ask why FileSystem is not initialized here too.
```

Comment:
The question why. If we see FileSystem as a service, we could put their initialization inside initializeServices. Then there will be no need to propagate pluginManager from the outside, it will be scoped to the same context.

Reply:
Again I'm not sure. Frankly I'm not very familiar with this portions of the code. One question that I would ask, is if `installSecurityContext()` can require an already initialized `FileSystem`?

If you would like to move the things around here, I would suggest to do it in separate PR in which we would need to involve Till probably.